### ISSUE TITLE
TypeError: nil can't be coerced into Float

### ISSUE THREAD

**Issue by sridharraman** [joshuaclayton/polylines#8]
I get this error when I run `Polylines::Decoder.decode_polyline(encoded_points)`. 

But, the same points works fine in Google's interactive decoder.

E.g. encoded_points = "epnyHvtb@`DuAfAl@t@nDzAnAnm@mq@yi@iqBdn@w_Bas@mUi[i[TcCfAq@`B\\qBe@r@_@aGbJ"

Any idea why this could be happening?

**Comment by donkey-hotei** [comment]
Would also like to confirm this bug when attempting to decode the polyline:
```
kbdoH{ioqCoF_j@Lwc@lC{BhNoMzCuJbEm]?wKkBaQmD}EqM_GwFiBfCge@zE_K~SaTdT`L|Da\\\\xJsX|EyJtQgPpXuRjKAtJqBfM[pn@g^nF}G??
```
Happens here:  https://github.com/joshuaclayton/polylines/blob/master/lib/polylines/decoder.rb#L13

**Comment by BuonOmo** [comment]
@donkey-hotei it looks like this works well with [`FastPolylines`][fast] at least! However, to debug such issue, we should also have the array of locations once decode.

@joshuaclayton I have more information on the issue, `transform_to_array_of_lat_lng_and_deltas` is returning an array of 55 elements, which should not be possible. And it seems like there should be 54 of those, hence a _quick and dirty_ fix would be:

```diff
diff --git a/lib/polylines/decoder.rb b/lib/polylines/decoder.rb
index f8479a4..eda432c 100644
--- a/lib/polylines/decoder.rb
+++ b/lib/polylines/decoder.rb
@@ -7,7 +7,7 @@ module Polylines
       [].tap do |points|
         points << [points_with_deltas.shift, points_with_deltas.shift]
 
-        while points_with_deltas.any?
+        while points_with_deltas.size > 1
           points << [
             points.last[0] + points_with_deltas.shift,
             points.last[1] + points_with_deltas.shift
```

[fast]: https://github.com/klaxit/fast-polylines

**Comment by joshuaclayton** [comment]
@BuonOmo thanks for pointing out the fix, and @sridharraman @donkey-hotei for highlighting the issue! I was able to capture the failure in a test, apply the change, and I've just [published 0.4.0](https://rubygems.org/gems/polylines/versions/0.4.0).
